cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/batch/mapreduce/MapReduceSinkContext.java

```diff
@@ -98,4 +96,14 @@ public Output call() throws Exception {
   public Set<String> getOutputNames() {
     return outputNames;
   }
+
+  private Output suffixOutput(Output output) {
+    // Create a unique local suffix string for alias
+    String uniqueSuffixCount = String.format("%s.%d", aliasSuffix, outputNames.size());
+    // If alias is null then, set it to uniqueSuffixCount, else suffix it with uniqueSuffixCount
```

Comment:
Hmm, seems like the null check should simply be in Input/Output classes then, and not here and in the SourceContext. It'll simplify this code that way.